mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -248,6 +269,11 @@ class KMeans @Since("1.5.0") (
       .setMaxIterations($(maxIter))
       .setSeed($(seed))
       .setEpsilon($(tol))
+
+    if (isSet(initialModel)) {
+      algo.setInitialModel($(initialModel).parentModel)
```

Comment:
does MLlibKMeans check if the dimensionalities match? Also, why do you set it to parentModel?